### PR TITLE
maint/dev ~ add EditorConfig

### PR DESCRIPTION
**Pull request by rivy**
Adds up-to-date EditorConfig.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+[*.{bash,sh}]
+# Linux/POSIX shell scripts
+indent_size = 4
+indent_style = space
+
+[*.{bat,cmd,[Bb][Aa][Tt],[Cc][Mm][Dd]}]
+# BAT/CMD ~ DOS/Win requires BAT/CMD files to have CRLF EOLNs
+end_of_line = crlf
+
+[*.{c,cc,cpp,cxx,h,hh,hpp,hxx,[Cc],[Cc][Cc],[Cc][Pp][Pp],[Cc][Xx][Xx],[Hh],[Hh][Hh],[Hh][Pp][Pp],[Hh][Xx][Xx]}]
```
Comment (sylvestre) [inline .editorconfig:35]: why bother to add the configuration for languages we won't use?

Comment (rivy) [reply .editorconfig:35]: It's really for development convenience. I try to keep one commented/"versioned" file for use where-ever I'm developing.
I can test vs my common config for any updates; it helps me keep up with changes and use the same practices throughout my projects.

I think it's useful without any real downside. Do you disagree/object?

In review of the file just now, I notice that I don't have TOML. I'm going to "DRAFT" for the moment until I add TOML, probably later today.

Comment (sylvestre) [reply .editorconfig:35]: I would prefer to keep only relevant information

**Comment by rivy** [comment]
Sure. It's a convenience for me, not a necessity.
We can close and reopen with some other version at a later date.

**Pull request closed without merging**